### PR TITLE
Update base terrain for deprecated *^Uf*

### PR DESCRIPTION
**Pull request by leonardoInf**
I mistakingly deleted my fork so I had to open this PR instead of https://github.com/wesnoth/wesnoth/pull/4838.

### CODE REVIEW

File: data/core/terrain-graphics.cfg [hunk 1]
```diff
@@ -888,7 +888,7 @@ C*,K*,X*,Q*,W*,Ai,M*,*^V*,*^B*,_off^_usr#enddef
 {NEW:TRANSITION            (Ms,Ha)          (W*,S*)                     -171               hills/snow-to-water}
 {NEW:TRANSITION            (Ms,Ha)          (!,Ha,Qx*,Mm,Ms,Md)         -172               hills/snow}
 
-{NEW:TRANSITION            (Tb)             (!,M*,Tb)                   -184               forest/mushroom-base}
+{NEW:TRANSITION            (Tb,*^Uf*)       (!,M*,Tb)                   -184               forest/mushroom-base}
```
Comment (stevecotton) [inline data/core/terrain-graphics.cfg:891]: `Tb` occurs in two places on this line, but `*^Uf*` only once - should it be like that?

Sorry for taking so long to start reviewing this, and I haven't started testing it in game yet.

Comment (leonardoInf) [reply data/core/terrain-graphics.cfg]: Yeah sure, it does not seem to do much but I have added it for consistency.

**Comment by leonardoInf** [comment]
There is a potential issue which is due to the fact that ^Uf* is Uu^Uf*, i.e. has the Uu terrain set in terrain.cfg. This seems to enables cave transitions and it shows the wrong preview when placing a ^Uf*. I do not think it is possible to change this without assigning ^Uf to Tb which would break backwards compability with legacy code. I think there is adecision to make here: Port ^Uf to Tb or accept the fact that it is not graphically equivalent to Tb unless there is a solution I am not aware of.

File: data/core/terrain-graphics.cfg (hunk 1 above)
Comment (leonardoInf) [reply data/core/terrain-graphics.cfg]: You can test it in the editor by setting commenting hide=yes in terrain.cfg

**Comment by leonardoInf** [comment]
TbvUu

This sums up the issue with sticking with Uu

**Comment by Pentarctagon** [comment]
Given that `^Uf` is deprecated and can't be selected directly in the editor anymore, the preview being incorrect sounds fairly unimportant.  The difference between the Uu^Uf and Tb^Uf also look pretty minor.

**Comment by leonardoInf** [comment]
@stevecotton What do you say. Is this acceptable?

**Comment by stevecotton** [comment]
Looks good to me, sorry that it took me so long to review.

**Comment by Pentarctagon** [comment]
Assuming nothing else is brought up then, I'll click the merge button tomorrow, unless someone else wants to do it first.

**Pull request merged**